src/storage/raw/store.rs

```diff
@@ -189,6 +189,9 @@ impl<'a, S: Snapshot> RawStoreInner<S> {
                     cursor.value(statistics).to_owned()
                 },
             )));
+            if pairs.len() >= limit {
```

Comment:
How about `if pairs.len() < limit { cursor.next() } else { break }`? The difference is the first branch is likely to hit.

Reply:
Sure! It is better, I will modify it.